apps/schools/domains/employee/components/employeeList/index.tsx

```diff
@@ -1,111 +1,47 @@
-import React, { useState, useEffect } from 'react'
-import { Table, Skeleton, Typography } from 'antd'
+import React, { useState } from 'react'
+import { Typography } from 'antd'
 import router from 'next/router'
 import styles from './styles/styles.module.scss'
 import { useGetAllEmployeesQuery } from '@domains/employee/redux/employeeApi'
-import { objectReBuilder, RawColumnType, useGenerateFullColumns } from '@domains/common/components/table'
 import { useOrganization } from '@domains/organization/providers/organizationProvider'
 import { Button } from '@domains/common/components/button'
-import { Input } from '@domains/common/components/input'
-import { SearchOutlined } from '@ant-design/icons'
-
-interface DataType {
-    id: string
-    name: string
-    position: string
-    phone: string
-}
-
-const baseColumns: RawColumnType<DataType>[] = [
-    {
-        dataIndex: 'id',
-        key: 'id',
-        hidden: true,
-    },
-    {
-        title: 'ФИО',
-        dataIndex: 'name',
-        key: 'name',
-        width: '30%',
-    },
-
-    {
-        title: 'Должность',
-        dataIndex: 'position',
-        key: 'position',
-        width: '30%',
-    },
-    {
-        title: 'Телефон',
-        dataIndex: 'phone',
-        key: 'phone',
-        width: '30%',
-    },
-].filter((item) => !item.hidden)
+import { searchColumns } from '@domains/common/constants/Table'
+import { Table } from '@domains/common/components/table'
+import { createSearchTextForRequest } from '@domains/common/utils/searchText'
 
 export function EmployeeList() {
-    const [dataSource, setDataSource] = useState<any[]>([])
+    const [searchRequestText, setSearchRequestText] = useState('')
```

Comment:
can we locate this logic to tableWithSearch component?

Reply:
It is not clear why, too much logic should be added to the component in order to replace useState with a constant. Keep It Simple, Stupid.